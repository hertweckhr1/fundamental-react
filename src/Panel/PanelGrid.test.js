import { mount } from 'enzyme';
import Panel from './Panel';
import PanelGrid from './PanelGrid';
import React from 'react';
import renderer from 'react-test-renderer';

describe('<PanelGrid />', () => {
    const panelGrid = (
        <PanelGrid className='blue'>
            <Panel colSpan={2}>
                <Panel.Head title={'Panel Header with Actions'} />
                <Panel.Body>Panel</Panel.Body>
            </Panel>
            <Panel>
                <Panel.Head description='Panel Description' />
                <Panel.Body>Panel</Panel.Body>
            </Panel>
            <Panel>
                <Panel.Head />
                <Panel.Body>Panel</Panel.Body>
            </Panel>
            <Panel>
                <Panel.Body>Panel</Panel.Body>
            </Panel>
            <Panel>
                <Panel.Body>Panel</Panel.Body>
            </Panel>
        </PanelGrid>
    );

    const panelGridNoGap = (
        <PanelGrid nogap>
            <Panel>
                <Panel.Body>Panel</Panel.Body>
            </Panel>
            <Panel>
                <Panel.Body>Panel</Panel.Body>
            </Panel>
            <Panel>
                <Panel.Body>Panel</Panel.Body>
            </Panel>
            <Panel>
                <Panel.Body>Panel</Panel.Body>
            </Panel>
            <Panel>
                <Panel.Body>Panel</Panel.Body>
            </Panel>
        </PanelGrid>
    );

    const panelGridSpan = (
        <PanelGrid cols={2}>
            <Panel>
                <Panel.Body>Panel</Panel.Body>
            </Panel>
            <Panel>
                <Panel.Body>Panel</Panel.Body>
            </Panel>
            <Panel>
                <Panel.Body>Panel</Panel.Body>
            </Panel>
            <Panel>
                <Panel.Body>Panel</Panel.Body>
            </Panel>
        </PanelGrid>
    );

    test('create panels', () => {
        // panel grid
        let component = renderer.create(panelGrid);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // panel with span
        component = renderer.create(panelGridSpan);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();

        // panel with no gap
        component = renderer.create(panelGridNoGap);
        tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    describe('Prop spreading', () => {
        test('should allow props to be spread to the PanelGrid component', () => {
            const element = mount(<PanelGrid data-sample='Sample' />);

            expect(
                element.getDOMNode().attributes['data-sample'].value
            ).toBe('Sample');
        });
    });
});
