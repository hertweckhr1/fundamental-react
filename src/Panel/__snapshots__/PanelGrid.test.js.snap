// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PanelGrid /> create panels 1`] = `
<div
  className="fd-panel-grid blue"
>
  <div
    className="fd-panel fd-has-grid-column-span-2"
  >
    <div
      className="fd-panel__head"
    >
      <h1
        className="fd-panel__title"
      >
        Panel Header with Actions
      </h1>
    </div>
    <div
      className="fd-panel__body"
    >
      Panel
    </div>
  </div>
  <div
    className="fd-panel"
  >
    <div
      className="fd-panel__head"
    >
      <p
        className="fd-panel__description"
      >
        Panel Description
      </p>
    </div>
    <div
      className="fd-panel__body"
    >
      Panel
    </div>
  </div>
  <div
    className="fd-panel"
  >
    <div
      className="fd-panel__head"
    />
    <div
      className="fd-panel__body"
    >
      Panel
    </div>
  </div>
  <div
    className="fd-panel"
  >
    <div
      className="fd-panel__body"
    >
      Panel
    </div>
  </div>
  <div
    className="fd-panel"
  >
    <div
      className="fd-panel__body"
    >
      Panel
    </div>
  </div>
</div>
`;

exports[`<PanelGrid /> create panels 2`] = `
<div
  className="fd-panel-grid fd-panel-grid--2col"
>
  <div
    className="fd-panel"
  >
    <div
      className="fd-panel__body"
    >
      Panel
    </div>
  </div>
  <div
    className="fd-panel"
  >
    <div
      className="fd-panel__body"
    >
      Panel
    </div>
  </div>
  <div
    className="fd-panel"
  >
    <div
      className="fd-panel__body"
    >
      Panel
    </div>
  </div>
  <div
    className="fd-panel"
  >
    <div
      className="fd-panel__body"
    >
      Panel
    </div>
  </div>
</div>
`;

exports[`<PanelGrid /> create panels 3`] = `
<div
  className="fd-panel-grid fd-panel-grid--nogap"
>
  <div
    className="fd-panel"
  >
    <div
      className="fd-panel__body"
    >
      Panel
    </div>
  </div>
  <div
    className="fd-panel"
  >
    <div
      className="fd-panel__body"
    >
      Panel
    </div>
  </div>
  <div
    className="fd-panel"
  >
    <div
      className="fd-panel__body"
    >
      Panel
    </div>
  </div>
  <div
    className="fd-panel"
  >
    <div
      className="fd-panel__body"
    >
      Panel
    </div>
  </div>
  <div
    className="fd-panel"
  >
    <div
      className="fd-panel__body"
    >
      Panel
    </div>
  </div>
</div>
`;
